### DISCUSSION TITLE
Add validation for o2m-type relations

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [directus/directus#7178] [Feature Requests]
We should be able to setup validation, like `required`, for o2m type fields. This includes specialized flags for things like "min/max number of items".

@benhaynes can you think through how to configure this in the field setup flow?

**Comment by benhaynes** [comment]
I know this is weird, but thoughts on saving that in `directus_relations` since the API would need to enforce it? Then it would be more appropriate to have under the relationship setup pane.

**Comment by benhaynes** [comment]
Did we just change our thinking on this? I like the idea of having this in the schema/relationship, though it also could go under Validation using a `COUNT()`. I'm wondering which makes more sense... @rijkvanzanten ?

**Comment by rijkvanzanten** [comment]
@benhaynes It's the question of data integrity vs permissions 🤔 Setting it on the relationship makes it kinda like "allow null" in a way where it's globally the same for everybody, while doing it on the validation step makes it more a "As a public user, you can only like 3 things at once"

**Comment by benhaynes** [comment]
Yeah, that's what I was thinking too. Both are useful, we can always extend permissions in the future with `COUNT()`, so we just need to decide if having a strict rule on the relationship is important too.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
